pkg/kubelet/kubelet_node_status.go

```diff
@@ -86,6 +90,7 @@ func (kl *Kubelet) registerWithAPIServer() {
 			kl.registrationCompleted = true
 			return
 		}
+		previousNode = node
```

Comment:
You are assigning `previousNode` to the output of `initialNode()` in line `77` which IIUC doesn't get updated in `tryRegisterWithAPIServer()`.
I think what you are looking for will be achieved if you were to make `tryRegisterWithAPIServer()` return old node object when it deletes it and assign that to `previous_node`. That way when you re-create the node object you'll have the extended resources intact and set to zero.

Reply:
I modified the code as you suggested. PTAL.